### PR TITLE
CNV-13866 and CNV-13122: New SNO content

### PR DESCRIPTION
**Pull request by sabrinajess**
Re: [CNV-13866](https://issues.redhat.com/browse/CNV-13866) and [CNV-13122](https://issues.redhat.com/browse/CNV-13122)
For 4.10 and later

Preview: https://deploy-preview-39384--osdocs.netlify.app/openshift-enterprise/latest/virt/install/preparing-cluster-for-virt
https://deploy-preview-39384--osdocs.netlify.app/openshift-enterprise/latest/virt/virtual_machines/virt-create-vms.html
https://deploy-preview-39384--osdocs.netlify.app/openshift-enterprise/latest/virt/vm_templates/virt-creating-vm-template.html

### CODE REVIEW

File: virt/virtual_machines/virt-create-vms.adoc [hunk 1]
```diff
@@ -22,6 +22,11 @@ When you create virtual machines from the web console, select a virtual machine
 
 Templates without a boot source are labeled as *Boot source required*. You can use these templates if you complete the steps for xref:../../virt/vm_templates/virt-creating-vm-template.adoc#virt-adding-a-boot-source-web_virt-creating-vm-template[adding a boot source to the virtual machine].
 
+[IMPORTANT]
+====
+You must remove the `evictionStrategy` from any templates or virtual machines that use `DataVolumes` and storage profiles to ensure compatibility with `SingleReplica` mode.
```
Comment (kgoldbla) [inline virt/virtual_machines/virt-create-vms.adoc:27]: Same here, to give context when this is relevant I would write it like this:

Due to differences in storage behavior, to ensure compatibility when creating a cluster with `SingleReplica` mode, you must remove the `evictionStrategy` from any templates or virtual machines that use `DataVolumes` and storage profiles.

**Review by kgoldbla — CHANGES_REQUESTED** [review]

File: modules/virt-about-vm-templates.adoc [hunk 2]
```diff
@@ -22,3 +22,8 @@ In the Templates tab, you cannot edit or delete Red Hat Supported or Red Hat Pro
 Using a Red Hat template is convenient because the template is already preconfigured. When you select a Red Hat template to create your own custom template, the *Create Virtual Machine Template* wizard prompts you to add a boot source if a boot source was not added previously. Then, you can either save your custom template or continue to customize it and save it.
 
 You can also select the *Create Virtual Machine Template* wizard directly and create a custom virtual machine template. The wizard prompts you to provide configuration details for the operating system, flavor, workload type, and other settings. You can add a boot source and continue to customize your template and save it.
+
+[IMPORTANT]
+====
+You must remove the `evictionStrategy` from any templates or virtual machines that use `DataVolumes` and storage profiles to ensure compatibility with `SingleReplica` mode.
```
[the remark this answers is not in the document]
Comment (sabrinajess) [reply modules/virt-about-vm-templates.adoc]: Updated admonition to reflect full text of bullet from Single Replica mode section.

File: virt/virtual_machines/virt-create-vms.adoc (hunk 1 above)
Comment (sabrinajess) [reply virt/virtual_machines/virt-create-vms.adoc]: Updated admonition to reflect full text of bullet from Single Replica mode section.

**Review by kgoldbla — APPROVED** [review]

**Comment by dominikholler** [comment]
FYI @borod108

**Comment by kbidarkar** [comment]
LGTM

File: modules/virt-using-singlereplica.adoc [hunk 3]
```diff
@@ -0,0 +1,12 @@
+// Module included in the following assemblies:
+//
+// * virt/install/preparing-cluster-for-virt.adoc
+
+[id="virt-using-singlereplica_{context}"]
+= Using SingleReplica mode
+
+Because the process of creating a cluster in `SingleReplica` mode or `HighlyAvailable` mode is determined at the time of installation, you do not need to configure anything specific to {VirtProductName} to use it on a single node. However; there are significant changes to the behavior of the cluster:
```
Comment (sjhala-ccs) [inline modules/virt-using-singlereplica.adoc:8]: Maybe reword the content to something like this:

```
If your cluster uses the `SingleReplica` mode, also known as Single Node OpenShift (SNO) (xref to OCP doc will go here), the following changes occur in {VirtProductName} behavior:

- {VirtProductName} components have only one replica and pod disruption budgets are not set.
- Live migration is not supported.
- Due to differences in storage behavior, some virtual machine templates are incompatible with `SingleReplica` mode. To ensure that the virtual machine is shut off in case of a node drain, you must not use the `evictionStrategy` field for any templates or virtual machines that use data volumes or storage profiles.
```

**Review by sjhala-ccs — COMMENTED** [review]
A few suggestions for you to consider, thanks Sabrina!

File: modules/virt-using-singlereplica.adoc (hunk 3 above)
Comment (sabrinajess) [reply modules/virt-using-singlereplica.adoc]: After reading your edits, I revisited convos with a couple different folks. I incorporated some of your feedback, along with some different rewrites, and opted to better mirror the OCP page. Kept in some extra details devs felt was relevant.

File: modules/virt-single-node-cluster.adoc [hunk 4]
```diff
@@ -0,0 +1,12 @@
+// Module included in the following assemblies:
+//
+// * virt/install/preparing-cluster-for-virt.adoc
+
+[id="virt-single-node-cluster_{context}"]
+= Changes in {VirtProductName} behavior on a single node cluster
+
+By default, you can use {VirtProductName} with single node clusters, also known as xref:../../operators/operator_sdk/osdk-ha-sno.adoc#osdk-ha-sno[Single Node OpenShift (SNO)]. However, these clusters are not configured for high-availability operation, which results in significant changes to the behavior of the cluster:
+
+* {VirtProductName} components have only one replica, and `PodDisruptionBudgets` are skipped.
```
Comment (sjhala-ccs) [inline modules/virt-single-node-cluster.adoc:10]: s/`PodDisruptionBudgets`/pod disruption budgets

Per our [API object formatting guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#api-object-formatting), you do not need to mark up general references to the object in backticks.

Comment (sjhala-ccs) [inline modules/virt-single-node-cluster.adoc:10]: **Suggestion:** 
s/are skipped/do not apply 
OR
s/are skipped/are not set

File: modules/virt-single-node-cluster.adoc [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+* Live migration is not available. There is no supported way to migrate a virtual machine to the same node that it is already on.
```
Comment (sjhala-ccs) [inline modules/virt-single-node-cluster.adoc:11]: Per minimalism principles, I think saying "Live migration is not supported" should be enough here.

File: modules/virt-single-node-cluster.adoc [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+* Due to differences in storage behavior, some virtual machine templates are incompatible with SNO. To ensure compatibility, you must remove the `evictionStrategy` from any templates or virtual machines that use data volumes or storage profiles.
```
Comment (sjhala-ccs) [inline modules/virt-single-node-cluster.adoc:12]: If this is a reference to the `evictionStrategy field in the API object, I would suggest rewording the second sentence as:

```
To ensure compatibility, do not set the `evictionStrategy` field for any templates or virtual machines that use data volumes or storage profiles.
```

File: virt/virtual_machines/virt-create-vms.adoc [hunk 7]
```diff
@@ -22,6 +22,11 @@ When you create virtual machines from the web console, select a virtual machine
 
 Templates without a boot source are labeled as *Boot source required*. You can use these templates if you complete the steps for xref:../../virt/vm_templates/virt-creating-vm-template.adoc#virt-adding-a-boot-source-web_virt-creating-vm-template[adding a boot source to the virtual machine].
 
+[IMPORTANT]
+====
+Due to differences in storage behavior, some virtual machine templates are incompatible with `SingleReplica` mode. You must remove the `evictionStrategy` from any templates or virtual machines that use `DataVolumes` and storage profiles to ensure compatibility with `SingleReplica` mode.
```
Comment (sjhala-ccs) [inline virt/virtual_machines/virt-create-vms.adoc:27]: Can you please rewrite this text to match the last bullet list item in the `virt-single-node-cluster` module? Same comment applies to the note in the `virt-about-vm-templates.adoc` module as well.

**Review by sjhala-ccs — COMMENTED** [review]
This looks great overall, just a few items to resolve.
